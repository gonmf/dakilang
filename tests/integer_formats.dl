# Different integer formats

number(121, 'decimal').
number(0172, 'octal').
number(0b1111011, 'binary').
number(0x7c, 'hexadecimal').

number(N, Format)?

abc(A, [1.20, 0.12, 120.0, 120, 0170, 0x78, 0b1111000]).

abc('ok', A)?

listing
