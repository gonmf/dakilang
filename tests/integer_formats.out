> # Different integer formats
>
> number(121, 'decimal').
> number(0172, 'octal').
> number(0b1111011, 'binary').
> number(0x7c, 'hexadecimal').
>
> number(N, Format)?
N = 121
Format = 'decimal'

N = 122
Format = 'octal'

N = 123
Format = 'binary'

N = 124
Format = 'hexadecimal'

>
> abc(A, [1.20, 0.12, 120.0, 120, 0170, 0x78, 0b1111000]).
>
> abc('ok', A)?
A = [1.2, 0.12, 120.0, 120, 120, 120, 120]

>
> listing
0: number(121, 'decimal').
1: number(122, 'octal').
2: number(123, 'binary').
3: number(124, 'hexadecimal').
4: abc(A, [1.2, 0.12, 120.0, 120, 120, 120, 120]).

